### PR TITLE
Refactor to be more Swift3y

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

File: LDSContent/Catalog.swift [hunk 1]
```diff
@@ -230,102 +230,97 @@ extension Catalog {
...
+            return try db.prepareCursor(ItemTable.table.join(LibraryItemTable.table, on: ItemTable.table[ItemTable.id] == LibraryItemTable.itemID).join(LibrarySectionTable.table, on: LibraryItemTable.librarySectionID == LibrarySectionTable.table[LibrarySectionTable.id]).filter(LibrarySectionTable.libraryCollectionID == libraryCollectionID && validPlatformIDs.contains(ItemTable.platformID)).order(LibraryItemTable.position)).map { try ItemTable.fromNamespacedRow($0) }
         } catch {
-            ErrorLogger.logError(error)
+            Log.error(error)
             return []
         }
     }
     
-    public func itemsWithURIsIn(_ uris: [String], languageID: Int64) -> [Item] {
+    public func items(uris: [String], languageID: Int64) -> [Item] {
```
Comment (nickmshelley) [inline LDSContent/Catalog.swift]: `itemsWith(urisIn uris: [String], languageID: Int64)`

File: LDSContent/Catalog.swift [hunk 2]
```diff
@@ -230,102 +230,97 @@ extension Catalog {
...
             return try db.prepareCursor(ItemTable.table.filter(ItemTable.sourceID == sourceID && validPlatformIDs.contains(ItemTable.platformID))).map { try ItemTable.fromRow($0) }
         } catch {
-            ErrorLogger.logError(error)
+            Log.error(error)
             return []
         }
     }
     
-    public func itemsWithIDsIn(_ ids: [Int64]) -> [Item] {
+    public func items(ids: [Int64], languageID: Int64? = nil) -> [Item] {
```
Comment (nickmshelley) [inline LDSContent/Catalog.swift]: `itemsWith(idsIn ids: [Int64], ...)`

File: LDSContent/Catalog+SubitemMetadata.swift [hunk 3]
```diff
@@ -25,47 +25,47 @@ public extension Catalog {
...
             }
         } catch {
-            ErrorLogger.logError(error)
+            Log.error(error)
             return nil
         }
     }
     
-    public func docIDForSubitemWithID(_ subitemID: Int64, itemID: Int64) -> String? {
+    public func docIDForSubitem(subitemID: Int64, itemID: Int64) -> String? {
```
Comment (nickmshelley) [inline LDSContent/Catalog+SubitemMetadata.swift]: `docIDForSubitemWith(...)`

**Comment by nickmshelley** [comment]
I made a few suggested name changes, but I stopped after a bit. Maybe just read through this again and make the names conform better: https://swift.org/documentation/api-design-guidelines/

File: LDSContent/Log.swift [hunk 4]
```diff
@@ -22,11 +22,17 @@
...
 
-public struct ErrorLogger {
+public struct Log {
+    static let logErrorsToConsole = true
+    
     public static var logErrorHandler: ((String, Error) -> Void)?
     
-    static func logError(_ error: Error, function: String = #function, line: Int = #line) {
-        let message = "\(function) (\(line))"
+    static func error(_ error: Error, file: String = #file, function: String = #function, line: Int = #line) {
```
Comment (nickmshelley) [inline LDSContent/Log.swift]: I think we should have these be the same in the frameworks/apps and I like the changes you made here, so I took what you have here to use in Authentication and Annotations, but this one is missing an optional description. If you want, you can change this to my updated version:

``` swift
public struct Log {
    public static let logErrorsToConsole = true
    public static var logErrorHandler: ((String, Error) -> Void)?
    
    static func error(_ error: Error, description: String? = nil, file: String = #file, function: String = #function, line: Int = #line) {
        let location = String(format: "[%@ %@ :%d]", file.components(separatedBy: "/").last ?? file, function, line)
        let message = description.flatMap { "\($0)\n\n\(location)" } ?? location
        logErrorHandler?(message, error)
        
        if logErrorsToConsole {
            print(message, error)
        }
    }
}

File: LDSContent/Log.swift [hunk 5]
```diff
@@ -22,11 +22,17 @@
 
 import Foundation
 
-public struct ErrorLogger {
+public struct Log {
+    static let logErrorsToConsole = true
```
Comment (nickmshelley) [inline LDSContent/Log.swift]: This should probably be public so clients can override it.

Comment (nickmshelley) [inline LDSContent/Log.swift]: Oh and a `var`... 😄

**Comment by stephanheilner** [comment]
I will redo and resubmit this following the swift guidelines.

**Pull request closed without merging**